### PR TITLE
Documented RCC and EXTI for STM32L100

### PR DESCRIPTION
**Pull request by MattCatz**
Adding documentation. Note this relies on pull request #60. It is presented as two separate pull request for simplicity. This serves to fully document the RCC and EXTI peripherals for the STM32L100. The only register that was not documented was the trimming register because I do not think I understand the documentation.

### CODE REVIEW

File: peripherals/exti/exti_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+EXTI:
+  IMR:
+    "MR*":
+      Masked: [0, "Interrupt request line is masked"]
+      Unmasked: [1, "Interrupt request line is masked"]
```
Comment (adamgreig) [inline peripherals/exti/exti_v1.yaml]: The description here should be "unmasked" not "masked".

File: peripherals/exti/exti_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+EXTI:
+  IMR:
+    "MR*":
+      Masked: [0, "Interrupt request line is masked"]
+      Unmasked: [1, "Interrupt request line is masked"]
+  EMR:
+    "MR*":
+      Masked: [0, "Interrupt request line is masked"]
+      Unmasked: [1, "Interrupt request line is masked"]
```
Comment (adamgreig) [inline peripherals/exti/exti_v1.yaml]: (same as IMR)

File: peripherals/rcc/rcc_l1.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      _read:
+        Unlocked: [0, "PPL unlocked"]
+        Locked: [1, "PPL locked"]
+    HSEBYP:
+      NotBypassed: [0, "HSE oscillator not bypassed"]
+      Bypassed: [1, "HSE oscillator bypassed"]
+    "*RDY":
+      _read:
+        NotReadied: [0, "The selected oscillator is not stable"]
+        Readied: [1, "The selected oscillator is not stable"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: These should be `NotReady` and `Ready` please. The description for `Ready` needs fixing too (it says "not stable").

File: peripherals/rcc/rcc_l1.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      DIV8: [3, "Division by 8"]
+      DIV16: [4, "Division by 16"]
+    MCOSEL:
+      NoClock: [0, "No clock"]
+      SYSCLK: [1, "SYSCLK clock selected"]
+      HSI: [2, "HSI oscillator clock selected"]
+      MSI: [3, "MSI oscillator clock selected"]
+      HSE: [4, "HSE oscillator clock selected"]
+      PLL: [5, "PLL clock selected"]
+      LSI: [6, "LSI oscillat or clock selected"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Extra space in "oscillat or"

File: peripherals/rcc/rcc_l1.yaml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      LSE: [7, "LSE oscillator clock selected"]
+    PLLDIV:
+      DIV2: [2, "PLLVCO / 2"]
+      DIV3: [3, "PLLVCO / 3"]
+      DIV4: [4, "PLLVCO / 4"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: These constants should be 1, 2, 3 respectively.

File: peripherals/rcc/rcc_l1.yaml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      DIV16: [11, "system clock divided by 16"]
+      DIV64: [12, "system clock divided by 64"]
+      DIV128: [13, "system clock divided by 128"]
+      DIV256: [14, "system clock divided by 256"]
+      DIV512: [15, "system clock divided by 512"]
+    SWS:
+      _read:
+        HSI: [0, "MSI oscillator used as system clock"]
+        MSI: [1, "HSI oscillator used as system clock"]
+        HSE: [2, "oscillator used as system clock"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Description missing "HSE"

File: peripherals/rcc/rcc_l1.yaml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        PLL: [3, "PLL used as system clock"]
+    SW:
+      HSI: [0, "MSI oscillator used as system clock"]
+      MSI: [1, "HSI oscillator used as system clock"]
+      HSE: [2, "oscillator used as system clock"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Description missing "HSE"

File: peripherals/rcc/rcc_l1.yaml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+  CIR:
+    CSSC:
+      _write:
+        Cleared: [1, "Clear selected interrupt"]
+    "*CSSC":
+      _write:
+        Cleared: [1, "Clear selected interrupt"]
+    "*RDYC":
+      _write:
+        Cleared: [1, "Clear selected interrupt"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Please use `Clear` for these three fields, since they're write-only.

File: peripherals/rcc/rcc_l1.yaml [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    LSECSSIE:
+      Disabled: [0, "LSE CSS interrupt disabled"]
+      Enabled: [1, "LSE CSS interrupt enabled"]
+    "*RDYIE":
+      Disabled: [0, "Disabling selected interrupt"]
+      Enabled: [1, "Enabling selected interrupt"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Perhaps just "Interrupt disabled" and "Interrupt enabled" would be better.

File: peripherals/rcc/rcc_l1.yaml [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      Enabled: [1, "The selected clock is enabled"]
+  "AHBLPENR":
+    "*EN":
+      Disabled: [0, "The selected clock is disabled"]
+      Enabled: [1, "The selected clock is enabled"]
+  CSR:
+    "*RSTF":
+      _read:
+        NoRest: [0, "No reset has occured"]
+        Rest: [1, "A reset has occured"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: Should be `NoReset` and `Reset`, instead of `NoRest`/`Rest`.

**Comment by MattCatz** [comment]
Thanks for proof reading!

File: peripherals/rcc/rcc_l1.yaml [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+      _read:
+        Unlocked: [0, "PPL unlocked"]
+        Locked: [1, "PPL locked"]
+    HSEBYP:
+      NotBypassed: [0, "HSE oscillator not bypassed"]
+      Bypassed: [1, "HSE oscillator bypassed"]
+    "*RDY":
+      _read:
+        NotReady: [0, "Oscillator is not stable"]
+        Ready: [1, "Oscillator is not stable"]
```
Comment (adamgreig) [inline peripherals/rcc/rcc_l1.yaml]: The description here still needs correcting to say `is stable`

**Comment by adamgreig** [comment]
Thanks for the patches!

**Pull request merged**